spec/models/step_spec.rb

```diff
@@ -38,4 +40,52 @@
       expect(step_item.errors).to have_key(:logic)
     end
   end
+  describe 'link reports' do
+    it 'should return nothing if there are no link reports yet' do
+      expect(step_item.broken_links).to be_nil
+    end
+
+    it 'should contain one item if there are link reports and at least one is broken' do
+      link_checker_api_get_batch(
+        id: 2,
+        links: [
+          {
+            "uri": "https://www.gov.uk/",
+            "status": "ok",
+            "checked": "2017-04-12T18:47:16Z",
+            "errors": [],
+            "warnings": [],
+            "problem_summary": "null",
+            "suggested_fix": "null"
+          },
+          {
+            "uri": "https://www.gov.uk/404",
+            "status": "broken",
+            "checked": "2017-04-12T16:30:39Z",
+            "errors": [
+              "Received 404 response from the server."
+            ],
+            "warnings": [],
+            "problem_summary": "404 error (page not found)",
+            "suggested_fix": ""
+          },
+          {
+            "uri": "https://www.gov.uk/search",
+            "status": "ok",
+            "checked": "2017-04-12T18:55:29Z",
+            "errors": [],
+            "warnings": [],
+            "problem_summary": "null",
+            "suggested_fix": "null"
+          }
+        ]
+        )
+      create(:link_report, batch_id: 1, step_id: step_item.id)
+      create(:link_report, completed: Time.now, batch_id: 2, step_id: step_item.id)
+      broken_link_report = step_item.broken_links.first
+      expect(broken_link_report.fetch('uri')).to eq "https://www.gov.uk/404"
+      expect(broken_link_report.fetch('problem_summary')).to eq "404 error (page not found)"
+      expect(step_item.broken_links.length).to eql 1
```

Comment:
I think these assertions could be split up amongst multiple tests, so it's clear what piece of test data relates to an expectation.

For example, why does the second link report have a completed time? How is that being used/tested?
I think this test only needs to check that the most recent link report is being queried if more than one link report exists. 

You could check the values of the broken links being returned in another test. For that, you only need to stub one link report.